api/wallet.go

```diff
@@ -25,125 +74,343 @@ func scanAmount(amount string) (types.Currency, bool) {
+		UnconfirmedOutgoingSiacoins: siacoinsOut,
+		UnconfirmedIncomingSiacoins: siacoinsIn,
+
+		SiafundBalance:      siafundBal,
+		SiacoinClaimBalance: siaclaimBal,
+	})
+}
+
+// walletHander handles API calls to /wallet.
+func (srv *Server) walletHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "" || req.Method == "GET" {
+		srv.walletHandlerGET(w, req)
+	} else {
+		writeError(w, "unrecognized method when calling /wallet", http.StatusBadRequest)
+	}
+}
+
+// walletCloseHandlerPUT handles a PUT request to /wallet/close.
+func (srv *Server) walletCloseHandlerPUT(w http.ResponseWriter, req *http.Request) {
+	err := srv.wallet.Close()
+	if err == nil {
+		writeSuccess(w)
+	} else {
+		writeError(w, err.Error(), http.StatusBadRequest)
+	}
+}
+
+// walletCloseHanlder handles API calls to /wallet/close.
+func (srv *Server) walletCloseHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "PUT" {
+		srv.walletCloseHandlerPUT(w, req)
+	} else {
+		writeError(w, "unrecognized method when calling /wallet/close", http.StatusBadRequest)
+	}
+}
+
+// walletHistoryHandlerGET handles a GET request to /wallet/history.
+func (srv *Server) walletHistoryHandlerGET(w http.ResponseWriter, req *http.Request) {
+	start, err := strconv.Atoi(req.FormValue("startHeight"))
 	if err != nil {
-		writeError(w, "Failed to get a coin address", http.StatusInternalServerError)
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	end, err := strconv.Atoi(req.FormValue("endHeight"))
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	// Since coinAddress is not a struct, we define one here so that writeJSON
-	// writes an object instead of a bare value. In addition, we transmit the
-	// coinAddress as a hex-encoded string rather than a byte array.
-	writeJSON(w, struct{ Address types.UnlockHash }{unlockConditions.UnlockHash()})
+	confirmedHistory, err := srv.wallet.History(types.BlockHeight(start), types.BlockHeight(end))
+	if err != nil {
+		writeError(w, "/walet/history [GET] Error:"+err.Error(), http.StatusBadRequest)
+		return
+	}
+	writeJSON(w, WalletHistoryGET{
+		UnconfirmedTransactions: srv.wallet.UnconfirmedHistory(),
+		ConfirmedTransactions:   confirmedHistory,
+	})
 }
 
-/*
-// walletMergeHandler handles the API call to merge a different wallet into the
-// current wallet.
-func (srv *Server) walletMergeHandler(w http.ResponseWriter, req *http.Request) {
-	// Scan the wallet file.
-	err := srv.wallet.MergeWallet(req.FormValue("walletfile"))
+// walletHistoryHandlerGETaddr handles a GET request to
+// /wallet/history/$(addr).
+func (srv *Server) walletHistoryHandlerGETaddr(w http.ResponseWriter, req *http.Request, addr types.UnlockHash) {
+	addrHistory, err := srv.wallet.AddressHistory(addr)
 	if err != nil {
-		writeError(w, err.Error(), http.StatusBadRequest)
+		writeError(w, "error after call to /wallet/history/$(addr): "+err.Error(), http.StatusBadRequest)
```

Comment:
In the documentation, and for the other stuff, I treat anything `/$(xxx)` as a separate call. Same goes for `/wallet/transaction/$(id)`.